src/netex_idf/lines.rs

```diff
@@ -158,6 +277,41 @@ fn make_networks_companies(
     Ok((networks, companies, map_line_network))
 }
 
+fn make_physical_and_commercial_modes(
+    lines_netex_idf: &CollectionWithId<LineNetexIDF>,
+) -> Result<(
+    CollectionWithId<PhysicalMode>,
+    CollectionWithId<CommercialMode>,
+)> {
+    let mut physical_modes = CollectionWithId::default();
+    let mut commercial_modes = CollectionWithId::default();
+    let modes: BTreeSet<_> = lines_netex_idf.values().map(|l| &l.mode).collect();
+    for m in modes {
+        let (physical_mode_id, physical_mode_name, commercial_mode_id, commercial_mode_name) =
```

Comment:
like before
this would never happen.
You already test if mode exist https://github.com/CanalTP/transit_model/pull/386/files#diff-dc95a9a382c6c3104f734c584c7468f5R80